boofuzz/monitors/network_monitor.py

```diff
@@ -0,0 +1,29 @@
+from .imonitor import IMonitor
+from . import pedrpc
+
+# Important: IMonitor needs to come *after* pedrpc.Client in the
+# Inheritance list for the method resolution order to produce
+# correct results.
+
+
+class NetworkMonitor(IMonitor, pedrpc.Client):
+    def __init__(self, host, port):
+        super(pedrpc.Client, self).__init__(host, port)
+
+    def pre_send(self, target=None, fuzz_data_logger=None, session=None):
+        return super(pedrpc.Client, self).pre_send(session.total_mutant_index)
```

Comment:
Shouldn't this be `super(IMonitor, self)...`?
Same for the following lines as well as ProcessMonitor.

Reply:
Nope, it should call pre_send on the pedrpc.Client base class, which in turn forwards it via RPC. The inheritance here is chosen specifically in a way that the MRO is ImplementingMonitor->IMonitor->pedrpc.Client so that abstract properties of methods will be intact in the base class even with pedrpcs catch-all methods. However, this means that the functions have to be called while explicitely stating the base class to use.

It's a bit weird because pedprc.Client doesn't even really implement these methods, but resolves them dynamically on the rpc server.

Reply:
Alright, but PyCharm and Pytype are never going to figure that out so we'll have to add ignores.
https://google.github.io/pytype/user_guide.html#silencing-errors

Optimal:
https://www.jetbrains.com/help/pycharm/disabling-and-enabling-inspections.html#suppressing-comments

Reply:
I've added the pytype stuff, but I don't really know which PyCharm analysis I have to surpress as I don't have PyCharm on hand. If you can provide me the error PyCharm raises, I'll add the annotations (or just push it yourself, I don't mind either way).